drivers/scheduler/k8s/k8s.go

```diff
@@ -591,6 +593,25 @@ func validateSpec(in interface{}) (interface{}, error) {
 	return nil, fmt.Errorf("unsupported object: %v", reflect.TypeOf(in))
 }
 
+// filterPureVolumesIfEnabled will return whether or not the given PVC is a Pure
+// direct access volume, only if PureVolumes is true. If PureVolumes is false, all
+// volumes are valid.
+func (k *K8s) filterPureVolumesIfEnabled(claim *v1.PersistentVolumeClaim) (bool, error) {
```

Comment:
yea this is a lot better way to handle it, I'd just make sure the resize volumes till work before merging it. But i dont see any problems assuming we converting the right storageclass. also, since we are here, is it better to have a filter for pure_file volumes and pure_block volumes ?

Reply:
I think having the same one is fine, we can do more specific checks later.